dask/dataframe/core.py

```diff
@@ -357,6 +357,39 @@ def _rebuild(self, dsk, *, rename=None):
     def _constructor(self):
         return new_dd_object
 
+    @property
+    def divisions(self):
+        """
+        Tuple of ``npartitions + 1`` values, in ascending order, marking the
+        lower/upper bounds of each partition's index. Divisions allow Dask
+        to know which partition will contain a given value, significantly
+        speeding up operations like `loc`, `merge`, and `groupby` by not
+        having to search the full dataset.
+
+        Example: for ``divisions = (0, 10, 50, 100)``, there are three partitions,
+        where the index in each partition contains values [0, 10), [10, 50),
+        and [50, 100), respectively. Dask therefore knows ``df.loc[45]``
+        will be in the second partition.
```

Comment:
I was recently working on divisions for dask-awkward and ran into this potentially tricky logic. For this example (`divisions = (0, 10, 50, 100)`) I think the last division would contains the values [50, 100]; `divisions[-1]` is the last index in the dataframe.

```python
In [32]: df = pd.DataFrame(np.zeros(shape=(9, 3)), columns=("a", "b", "c"))

In [33]: df
Out[33]: 
     a    b    c
0  0.0  0.0  0.0
1  0.0  0.0  0.0
2  0.0  0.0  0.0
3  0.0  0.0  0.0
4  0.0  0.0  0.0
5  0.0  0.0  0.0
6  0.0  0.0  0.0
7  0.0  0.0  0.0
8  0.0  0.0  0.0

In [34]: df = dd.from_pandas(df, npartitions=3)

In [35]: df
Out[35]: 
Dask DataFrame Structure:
                     a        b        c
npartitions=3                           
0              float64  float64  float64
3                  ...      ...      ...
6                  ...      ...      ...
8                  ...      ...      ...
Dask Name: from_pandas, 3 tasks

In [36]: df.divisions
Out[36]: (0, 3, 6, 8)
```